drivers/SmartThings/zwave-switch/src/fibaro-double-switch/init.lua

```diff
@@ -53,45 +59,93 @@ local function central_scene_notification_handler(self, device, cmd)
   elseif cmd.args.key_attributes == 4 then
     button_number = cmd.args.scene_number + 4
   end
-
   local component = device.profile.components["button" .. button_number]
-
+  
   if component ~= nil then
     device:emit_component_event(component, event({state_change = true}))
   end
 end
 
-local function component_to_endpoint(device, component_id)
-  if component_id == "main" then
-    return {1}
-  else
-    return {2}
+local function device_added(driver, device, event)
+  if device.network_type == st_device.NETWORK_TYPE_ZWAVE then
+    local name = string.format("%s %s", device.label, "(CH2)")
+    local metadata = {
+      type = "EDGE_CHILD",
+      label = name,
+      profile = "metering-switch",
+      parent_device_id = device.id,
+      parent_assigned_child_key = string.format("%02X", 2),
+      vendor_provided_label = name,
+    }
+    driver:try_create_device(metadata)
   end
 end
 
-local function endpoint_to_component(device, ep)
-  local switch_comp = string.format("switch%d", ep - 1)
-  if device.profile.components[switch_comp] ~= nil then
-    return switch_comp
+local function find_child(parent, ep_id)
+  if ep_id == 1 then
+    return parent
   else
-    return "main"
+    return parent:get_child_by_parent_assigned_key(string.format("%02X", ep_id))
   end
 end
 
-local device_init = function(self, device)
-  device:set_component_to_endpoint_fn(component_to_endpoint)
-  device:set_endpoint_to_component_fn(endpoint_to_component)
+local function endpoint_to_component(device, endpoint)
+  return "main"
+end
+
+local function component_to_endpoint(device, component)
+  return { 1 }
+end
+
+local function device_init(driver, device, event)
+  if device.network_type == st_device.NETWORK_TYPE_ZWAVE then
+    device:set_find_child(find_child)
+    device:set_endpoint_to_component_fn(endpoint_to_component)
+    device:set_component_to_endpoint_fn(component_to_endpoint)
+  end
+end
+
+local function do_refresh(driver, device, command)
+  if device:is_cc_supported(cc.SWITCH_BINARY) then
+    device:send_to_component(SwitchBinary:Get({}), command.component)
+  elseif device:is_cc_supported(cc.BASIC) then
+    device:send_to_component(Basic:Get({}), command.component)
+  end
+  if device:supports_capability_by_id(capabilities.powerMeter.ID) or device:supports_capability_by_id(capabilities.energyMeter.ID) then
+    device:send_to_component(Meter:Get({ scale = Meter.scale.electric_meter.WATTS }), command.component)
+    device:send_to_component(Meter:Get({ scale = Meter.scale.electric_meter.KILOWATT_HOURS }), command.component)
+  end
```

Comment:
Do You need those checks `device:supports_capability_by_id` since there is only one device that falls under this sub-driver ?

Reply:
fixed